agent/consul/fsm/commands_oss.go

```diff
@@ -107,41 +107,66 @@ var CommandsSummaries = []prometheus.SummaryDefinition{
+type entry struct {
+	msgType structs.MessageType
+	cmd     unboundCommand
+}
+
+func RegisteredCommands(fsm *FSM) map[structs.MessageType]command {
+	commands := make(map[structs.MessageType]command)
+	registerCommands(commands, fsm,
+		entry{structs.RegisterRequestType, (*FSM).applyRegister},
+		entry{structs.DeregisterRequestType, (*FSM).applyDeregister},
+		entry{structs.KVSRequestType, (*FSM).applyKVSOperation},
+		entry{structs.SessionRequestType, (*FSM).applySessionOperation},
+		// DEPRECATED (ACL-Legacy-Compat) - Only needed for v1 ACL compat
+		entry{structs.DeprecatedACLRequestType, (*FSM).deprecatedApplyACLOperation},
+		entry{structs.TombstoneRequestType, (*FSM).applyTombstoneOperation},
+		entry{structs.CoordinateBatchUpdateType, (*FSM).applyCoordinateBatchUpdate},
+		entry{structs.PreparedQueryRequestType, (*FSM).applyPreparedQueryOperation},
+		entry{structs.TxnRequestType, (*FSM).applyTxn},
+		entry{structs.AutopilotRequestType, (*FSM).applyAutopilotUpdate},
+		entry{structs.IntentionRequestType, (*FSM).applyIntentionOperation},
+		entry{structs.ConnectCARequestType, (*FSM).applyConnectCAOperation},
+		entry{structs.ACLTokenSetRequestType, (*FSM).applyACLTokenSetOperation},
+		entry{structs.ACLTokenDeleteRequestType, (*FSM).applyACLTokenDeleteOperation},
+		entry{structs.ACLBootstrapRequestType, (*FSM).applyACLTokenBootstrap},
+		entry{structs.ACLPolicySetRequestType, (*FSM).applyACLPolicySetOperation},
+		entry{structs.ACLPolicyDeleteRequestType, (*FSM).applyACLPolicyDeleteOperation},
+		entry{structs.ConnectCALeafRequestType, (*FSM).applyConnectCALeafOperation},
+		entry{structs.ConfigEntryRequestType, (*FSM).applyConfigEntryOperation},
+		entry{structs.ACLRoleSetRequestType, (*FSM).applyACLRoleSetOperation},
+		entry{structs.ACLRoleDeleteRequestType, (*FSM).applyACLRoleDeleteOperation},
+		entry{structs.ACLBindingRuleSetRequestType, (*FSM).applyACLBindingRuleSetOperation},
+		entry{structs.ACLBindingRuleDeleteRequestType, (*FSM).applyACLBindingRuleDeleteOperation},
+		entry{structs.ACLAuthMethodSetRequestType, (*FSM).applyACLAuthMethodSetOperation},
+		entry{structs.ACLAuthMethodDeleteRequestType, (*FSM).applyACLAuthMethodDeleteOperation},
+		entry{structs.FederationStateRequestType, (*FSM).applyFederationStateOperation},
+		entry{structs.SystemMetadataRequestType, (*FSM).applySystemMetadataOperation},
+		entry{structs.PeeringWriteType, (*FSM).applyPeeringWrite},
+		entry{structs.PeeringDeleteType, (*FSM).applyPeeringDelete},
+		entry{structs.PeeringTerminateByIDType, (*FSM).applyPeeringTerminate},
+		entry{structs.PeeringTrustBundleWriteType, (*FSM).applyPeeringTrustBundleWrite},
+		entry{structs.PeeringTrustBundleDeleteType, (*FSM).applyPeeringTrustBundleDelete},
+		entry{structs.PeeringSecretsWriteType, (*FSM).applyPeeringSecretsWrite},
+	)
+	registerEnterpriseCommands(commands, fsm)
+	return commands
+}
+
+func registerEnterpriseCommands(registry map[structs.MessageType]command, fsm *FSM) {}
```

Comment:
Could you move most of this file to a new file called `commands.go` and leave the empty `registerEnterpriseCommands` in `commands_oss.go` with this directive at the top:
```
//go:build !consulent
```
On the enterprise side we have another file called `commands_ent.go` which will have the enterprise implementation.